lib/access-context.js

```diff
@@ -32,9 +32,12 @@ function AccessContext(context) {
   }
   context = context || {};
 
+  assert(context.registry,
+    'Application registry is mandatory in AccessContext but missing in provided context');
```

Comment:
> IMO we should only fallback to loopback.registry in case localRegistry is set to false, or not set and running a loopback version where it defaults to false.

Sure, that sounds good too. It may be best to leave this change out of scope of this pull request, so that this patch can be finally landed?